components/chart/src/tooltip/chart-tooltip-container.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Component,
+  ChangeDetectorRef,
+  ElementRef,
+  EmbeddedViewRef,
+  NgZone,
+  OnDestroy,
+  ViewChild,
+} from '@angular/core';
+import {
+  AnimationEvent,
+  trigger,
+  state,
+  style,
+  transition,
+  animate,
+} from '@angular/animations';
+
+import { DT_CHART_TOOLTIP_FADE_TIME } from './chart-tooltip-config';
+import {
+  mixinNotifyDomExit,
+  HasNgZone,
+  CanNotifyOnExit,
+} from '@dynatrace/barista-components/core';
+import { Subject } from 'rxjs';
+import { CdkPortalOutlet, TemplatePortal } from '@angular/cdk/portal';
+
+// Boilerplate for applying mixins to DtChartTooltipContainer.
+export class DtChartTooltipContainerBase implements HasNgZone {
+  constructor(public _ngZone: NgZone) {}
+}
+export const _DtChartTooltipContainerMixin = mixinNotifyDomExit(
+  DtChartTooltipContainerBase,
+);
+
+@Component({
+  selector: 'dt-chart-tooltip-container',
+  template: '<ng-template cdkPortalOutlet></ng-template>',
+  styleUrls: ['chart-tooltip-container.scss'],
+  host: {
+    class: 'dt-chart-tooltip-container',
+    '[@fade]': '_animationState',
+    '(@fade.done)': '_animationDone($event)',
+  },
+  animations: [
```

Comment:
please create a reusable function for that that is located under `components/core/src/animations`. There are already some animation timings. 

Because then you can reuse the fade animation of the confirmation dialog  `components/confirmation-dialog/src/confirmation-dialog-state/confirmation-dialog-state.ts` That uses the same animation.

`fade-animation.ts`
```typescript

interface DtFadeAnimationConfiguration {
  /** Fading duration in milliseconds */
  fadeTime: number;
  /** animation trigger name */
  trigger?: string;
}

export const dtFadeAnimation = (config: DtFadeAnimationConfiguration): AnimationTriggerMetadata =>
  trigger(config.trigger || 'fade', [
    state('void', style({ opacity: 0 })),
    transition(
      ':enter',
      animate(`${config.fadeTime}ms ease-in-out`),
    ),
    transition(
      ':leave',
      animate(`${config.fadeTime}ms ease-in-out`),
    ),
  ]);

```

usage:

```
  host: {
    class: 'dt-chart-tooltip-container',
    '[@fade]': '_animationState',
    '(@fade.done)': '_animationDone($event)',
  },
  animations: [
    dtFadeAnimation({fadeTime: DT_CHART_TOOLTIP_FADE_TIME})
  ]
})
export class DtChartTooltipContainer extends _DtChartTooltipContainerMixin
```